tests/e2e/streams/streams_tests.py

```diff
@@ -408,19 +472,84 @@ def test_bootstrap_server(producer, topics, connection, transformation):
 
     for topic in topics:
         common.check_vertex_exists_with_topic_and_payload(
-            cursor, topic, SIMPLE_MSG)
+            cursor, topic, SIMPLE_MSG
+        )
 
 
 @pytest.mark.parametrize("transformation", TRANSFORMATIONS_TO_CHECK)
 def test_bootstrap_server_empty(producer, topics, connection, transformation):
     assert len(topics) > 0
     cursor = connection.cursor()
     with pytest.raises(mgclient.DatabaseError):
-        common.execute_and_fetch_all(cursor,
-                                     "CREATE STREAM test "
-                                     f"TOPICS {','.join(topics)} "
-                                     f"TRANSFORM {transformation} "
-                                     "BOOTSTRAP_SERVERS ''")
+        common.execute_and_fetch_all(
+            cursor,
+            "CREATE STREAM test "
+            f"TOPICS {','.join(topics)} "
+            f"TRANSFORM {transformation} "
+            "BOOTSTRAP_SERVERS ''",
+        )
+
+
+@pytest.mark.parametrize("transformation", TRANSFORMATIONS_TO_CHECK)
+def test_set_offset(producer, topics, connection, transformation):
+    assert len(topics) > 0
+    cursor = connection.cursor()
+    common.execute_and_fetch_all(
+        cursor,
+        "CREATE STREAM test "
+        f"TOPICS {topics[0]} "
+        f"TRANSFORM {transformation} "
+        "BATCH_SIZE 1",
+    )
+
+    messages = [f"{i} message" for i in range(1, 21)]
+
+    for message in messages:
+        producer.send(topics[0], message.encode()).get(timeout=60)
+
+    def consume():
+        common.start_stream(cursor, "test")
+
+        time.sleep(2)
+        common.stop_stream(cursor, "test")
+        res = common.execute_and_fetch_all(
+            cursor, "MATCH (n) RETURN n.payload"
+        )
+        return res
+
+    def execute_set_offset_and_consume(id):
+        common.execute_and_fetch_all(
+            cursor, f"CALL mg.kafka_set_stream_offset('test', {id})"
+        )
+        return consume()
+
+    with pytest.raises(mgclient.DatabaseError):
+        res = common.execute_and_fetch_all(
+            cursor, "CALL mg.kafka_set_stream_offset('foo', 10)"
+        )
+
+    res = execute_set_offset_and_consume(10)
+    assert len(res) == 10
+
+    def comparison_check(a, b):
+        return a == str(b).strip("'(,)")
+
+    assert all([comparison_check(a, b) for a, b in zip(messages[10:], res)])
```

Comment:
```suggestion
    assert all(comparison_check(a, b) for a, b in zip(messages[10:], res))
```
Should work I think.

Reply:
I changed this code based on Benjo's suggestion